### DISCUSSION TITLE
Pod annotation disappears after the pod restart/recreation. What is the best approach to recreate?

### DISCUSSION THREAD

**Discussion by ccna2ccda** [vmware-tanzu/velero#8300] [Community support Q&A]
Hi everyone,

Using the annotation command below, we can exclude pod volume to be backed up by velero; but the annotation will be removed after the pod get restarted. What is the best approach to attach this command to pod's lifecycle?  Assuming we can't change the original yaml/helmchart. Thanks.

_`kubectl -n YOUR_POD_NAMESPACE annotate pod/YOUR_POD_NAME backup.velero.io/backup-volumes-excludes=YOUR_VOLUME_NAME_1,YOUR_VOLUME_NAME_2,...`_

**Comment by kaovilai** [comment]
add it to whoever owns/create the new pod. ie. `Deployments` or `Replicasets` or something else which will contain `.spec.PodSpec.annotations` or similar.

**Reply by ccna2ccda** [reply]
Thanks for the quick reply. The issue with adding the annotation to the statefulset/deployment's pod spec is it will restart the pods immediately. 

when we schedule the backup, is it possible to add pre-start shell script to check if the annotation exists, if not, then run kubectl command to add them?  If the backup is scheduled to run 4 hours, then this script is executed as part of this schedule.

**Reply by sseago** [reply]
@ccna2ccda That is not currently possible, but once we implement https://github.com/vmware-tanzu/velero/issues/4067 that should allow this.

The pod restart issue may be a bootstrapping/one-time problem to get over. If you have general policies around which pods you want to exclude from filesystem backup, then you could set these annotations on the pod template spec at creation time. Also, depending on how you decide which pods you want to exclude from fs-backup, the "snapshot" volume policy has the same effect as backup-volumes-excludes -- matching volumes will use snapshot rather than filesystem backup, so you might be able to use that.

**Reply by ccna2ccda** [reply]
Thank you for the feedback
